Build failure: alpine

## Steps To Reproduce

Steps to reproduce the behavior:

1. build alpine

## Build log

<details>

<summary>Build Log</summary>

```
adrbklib.c: In function 'sort_addr_list':
adrbklib.c:4705:64: error: passing argument 4 of 'qsort' from incompatible pointer type [-Wincompatible-pointer-types]
 4705 |     qsort((qsort_t *)list, (size_t)(p - list), sizeof(char *), cmp_addr);
      |                                                                ^~~~~~~~
      |                                                                |
      |                                                                int (*)(const char *, const char *)
In file included from ../include/system.h:42,
                 from ../pith/headers.h:25,
                 from adrbklib.c:14:
/nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdlib.h:971:34: note: expected '__compar_fn_t' {aka 'int (*)(const void *, const void *)'} but argument is of type 'int (*)(const char *, const char *)'
  971 |                    __compar_fn_t __compar) __nonnull ((1, 4));
      |                    ~~~~~~~~~~~~~~^~~~~~~~
adrbklib.c: In function 'adrbk_sort':
adrbklib.c:4766:77: error: passing argument 4 of 'qsort' from incompatible pointer type [-Wincompatible-pointer-types]
 4765 |             (ab->sort_rule == AB_SORT_RULE_FULL_LISTS) ?
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                     
 4766 |                                                 cmp_cntr_by_full_lists_last :
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
      |                                                                             |
      |                                                                             int (*)(const char *, const char *)
 4767 |             (ab->sort_rule == AB_SORT_RULE_FULL) ?
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                           
 4768 |                                                 cmp_cntr_by_full :
      |                                                 ~~~~~~~~~~~~~~~~~~           
 4769 |             (ab->sort_rule == AB_SORT_RULE_NICK_LISTS) ?
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                     
 4770 |                                                 cmp_cntr_by_nick_lists_last :
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 4771 |             /* (ab->sort_rule == AB_SORT_RULE_NICK) */
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                       
 4772 |                                                 cmp_cntr_by_nick);
      |                                                 ~~~~~~~~~~~~~~~~  
```

</details>

## Additional context

`alpine` fails to build due to `-Wincompatible-pointer-types`. If I am not wrong, it was enabled by default with GCC 14.
It was fixed in Debian : https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1074804

## Metadata

```
$ nix-shell -p nix-info --run "nix-info -m"                                              
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.69, NixOS, 25.05 (Warbler), 25.05.20250108.bffc22e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/v0g0bxsd5gw6k0jz2855f8h7l1218925-source`
```

## Notify maintainers

@rhendric 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/al/alpine/package.nix
+++ b/pkgs/by-name/al/alpine/package.nix
@@ -43,6 +43,10 @@ stdenv.mkDerivation rec {
     "--with-c-client-target=slx"
   ];
 
+  # Fixes https://github.com/NixOS/nixpkgs/issues/372699
+  # See also https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1074804
+  env.NIX_CFLAGS_COMPILE = toString [ "-Wno-incompatible-pointer-types" ];
+
   passthru.updateScript = gitUpdater { rev-prefix = "v"; };
 
   meta = with lib; {

